source/write/transactions.txt

```diff
@@ -0,0 +1,192 @@
+.. _c-write-transactions:
+
+============
+Transactions
+============
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: ACID, write, consistency, code example
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to use the {+driver-short+} to perform
+**transactions**. Transactions allow you to perform a series of operations
+that change data only if the entire transaction is committed.
+If any operation in the transaction does not succeed, the library stops the
+transaction and discards all data changes before they ever become
+visible. This feature is called **atomicity**.
+
+In MongoDB, transactions run within logical **sessions**. A
+session is a grouping of related read or write operations that you intend to run 
+sequentially. Sessions enable **causal consistency** for a
+group of operations and allow you to run operations in an
+**ACID-compliant transaction**, which is a transaction that meets an expectation
+of atomicity, consistency, isolation, and durability. MongoDB guarantees that the 
+data involved in your transaction operations remains consistent, even if the 
+operations encounter unexpected errors.
+
+When using the {+driver-short+}, you can create a new session from a
+``mongoc_client_t`` instance. Then, you can use the resulting ``mongoc_client_session_t``
+instance to perform transactions.
+
+.. warning::
+
+   Use a ``mongoc_client_session_t`` only with the ``mongoc_client_t`` (or associated
+   ``mongoc_database_t`` or ``mongoc_collection_t``) that created it. Using a
+   ``mongoc_client_session_t`` with a different ``mongoc_client_t`` results in operation
+   errors.
+
+Transaction APIs
+----------------
+
+In this section, you can learn about the transaction APIs provided by
+the {+driver-short+}. Before you begin a transaction, you must create a
+``mongoc_client_session_t`` by using the ``mongoc_client_start_session()``
+function on your ``mongoc_client_t`` instance. Then, you can use either of the
+following APIs to perform a transaction:
+
+- :ref:`c-convenient-txn`
+- :ref:`c-core-txn`
+
+.. _c-convenient-txn:
+
+Convenient API
+~~~~~~~~~~~~~~
+
+The {+driver-short+} provides a **Convenient Transaction API** to manage
+the transaction lifecyle. Implement this API by using the
+``mongoc_client_session_with_transaction()`` function to run custom callback within a
+transaction. The ``mongoc_client_session_with_transaction()`` function performs the following
+tasks:
+
+- Starts the transaction
+- Handles errors by either ending the transaction or retrying it
```

Comment:
S: maybe you can be more specific about what kinds of errors lead to retry vs failure